build/xtask/src/config.rs

```diff
@@ -74,6 +74,7 @@ impl Config {
             toml::from_slice(&chip_contents)?
         };
 
+        assert!(!toml.tasks.contains_key("kernel"));
```

Comment:
This seems like kind of a random place for this, should it go higher? Should it `bail!` instead with a useful error message, such as `"sorry, you can't name a task kernel in the current system"` ?

Reply:
Sure, moved it to right after the `RawConfig` is loaded and switched to a more useful error message (`'kernel' is reserved and cannot be used as a task name`)